cmd/end2endtest/helpers.go

```diff
@@ -536,28 +537,43 @@ func (t *e2eElection) sendVotes(votes []*apiclient.VoteData) (errs map[int]error
 						delete(queue, i)
 					case errors.Is(err, context.DeadlineExceeded) || os.IsTimeout(err):
 						// if the context deadline is reached, no need to print it, just retry
-						timeouts.Add(1)
+						if timeoutsRetry.Load() > 5 {
+							errs = append(errs, err)
+							wg.Done()
```

Comment:
why this `wg.Done()`? there's already a `defer wg.Done()`.
i guess you wanted to `return` instead?

Reply:
Yeah, thanks